VALIDATION.md

```diff
@@ -0,0 +1,41 @@
+| `world.bg_effect` | A string that represents the current background effect | | &check;  |
+| `world.fg_effect` | A string that represents the current foreground effect | | &check;  |
+| `world.keysPressed` | A set that is assigned strings that represent keys pressed during the running of a user's program | &check; |  &check; |
+| `world.spriteGroupsCalledToChangeMove` | An array of strings that represent sprite groups called to change dance move via `changeMoveEachLR` | | &check;  |
+| `world.spriteStyles` | An array of strings that represent the sprite costumes that have been added to a user's program | | &check;  |
+| `world.validationState` | Object that is used to save state during the running of the user's program for validation purposes, [e.g.](https://github.com/code-dot-org/code-dot-org/blob/34be28fcefbf84f2cefec08e06abff311fc1befa/dashboard/config/levels/custom/dance/Dance_2019_03.level#L34C26-L34C26), `World.validationState[index] = sprite.getAnimationLabel()` | &check; |  &check; |
+| `getTime('measures')` | A function that returns a number | &check; |  &check; |
+| `setFuncContext(type, key)` | A function defined in `p5.dance.js` and added to `api.js` - this function is called on in `executeFuncs` defined in `p5.dance.interpreted` to set a function's context eventType and event key | | &check;  |
+| `sprites` | An array of sprites in a user's program - defined in `api.js` | &check; |  &check; |
+| `sprite.current_move` | A number that represents the index of a sprite's current dance move in [`MOVE_NAMES` array](https://github.com/code-dot-org/dance-party/blob/e5dc8698e8700314959e58cb8f45c6a3314c43cc/src/constants.js#L18) | &check; | |
+| `sprite.getAnimationLabel()` | A function that returns a string that represents a sprite's current animation, e.g., `anim0` - this function is defined in [`code-dot-org/p5.play`](https://github.com/code-dot-org/p5.play/blob/6b9a6ac479ce38a134cfc2fb9cadd50310741669/lib/p5.play.js#L2949C3-L2949C3) | &check; |  &check; |
+| `sprite.scale` | A number that represents the scale of a sprite - `0.3` is the default value for a backup dance and `1.0` is the default value for a lead dancer | &check; | |
+| `sprite.tint` | A string that represent the color of a sprite | &check; | |
+
+Note that `world` variables are accessed in `.level` files with `World`, e.g., `World.spriteStyles`, and functions defined in `p5.dance.js` are accessed in `.level` files with `nativeAPI`, e.g., `nativeAPI.pass()`. Functions and variables defined in `api.js` are accessed as is, e.g., `sprites`.
+
+Validation code is usually configured via levelbuilder by content editors. However, Dance Party validation code has been written by engineers because of the lack of documentation and the unfriendly API.
+
+[Link to Dance Party 2019 validation code](https://docs.google.com/document/d/1rZrUCeuy6KIRC8a44reu6QpNTCA_gCxHeDX6IFal96g/edit)  
```

Comment:
Thank you for writing these docs! 

I'm a little confused looking at these google docs links - are these in google docs cause they're collected from a bunch of different files in the codebase (or stored somewhere else?), just as examples for folks to look at? (sorry, you might have explained this to be already but I forgot >.>) This might benefit from some more explanation / pointing to an example of where these bits of code are stored for production use?

Reply:
Good call! I will add an explanation that I compiled the validation code from the .level files. Thanks!

Reply:
Much clearer! Thanks :)